extensions/crossmodel-lang/src/model-server/model-service.ts

```diff
@@ -92,21 +94,24 @@ export class ModelService {
     * If the document was not already open for modification, it will be opened automatically.
     *
     * @param uri document URI
+    * @param state minimum state the document should have before returning
     */
-   request(uri: string): Promise<AstNode | undefined>;
-   /**
-    * Requests the semantic model stored in the document with the given URI if it matches the given guard function.
-    * If the document was not already open for modification, it will be opened automatically.
-    *
-    * @param uri document URI
-    * @param guard guard function to ensure a certain type of semantic model
-    */
-   request<T extends AstNode>(uri: string, guard: (item: unknown) => item is T): Promise<T | undefined>;
-   async request<T extends AstNode>(uri: string, guard?: (item: unknown) => item is T): Promise<AstNode | T | undefined> {
+   async request(uri: string, state = DocumentState.Validated): Promise<AstCrossModelDocument | undefined> {
       const document = this.documents.getOrCreateDocument(URI.parse(uri));
+      // should be replaced once we upgrade to newer Langium versions so we can use documentBuilder.waitUntil or the waitUntilPhase function
```

Comment:
Should we prefix this comment with todo? So we can find it later in a todo list or something?

Or what is convention in JavaScript world for this? 🤔

Reply:
I'll tackle this with the next PR if that is fine with you since that already is the Langium upgrade ;-)

Reply:
In general pre-fixing with `TODO:` is better of course but it is already resolved in https://github.com/CrossBreezeNL/crossmodel/pull/64/files#diff-bfc5394600f3b18d3c7ec47fd842312acb8f3072dbf5a142c602b0d3e87ac792L101